include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -115,82 +109,92 @@ struct WorkSpace {
   //
   //        post_sorted <--- matmul
   //
-  Matrix<SizeType, Device::CPU> i1;
-  Matrix<SizeType, Device::CPU> i2;
-  Matrix<SizeType, Device::CPU> i3;
+  Matrix<SizeType, device> i1;
+  Matrix<SizeType, device> i2;
+  Matrix<SizeType, device> i3;
 
   // Assigns a type to each column of Q which is used to calculate the permutation indices for Q and U
   // that bring them in matrix multiplication form.
-  Matrix<ColType, Device::CPU> c;
+  Matrix<ColType, device> c;
+};
+
+// Device::GPU
+template <class T, Device D>
+struct WorkSpaceHostMirror {
+  matrix::Matrix<T, Device::CPU> evals;
+  matrix::Matrix<T, Device::CPU> mat1;
+  matrix::Matrix<T, Device::CPU> dtmp;
+  matrix::Matrix<T, Device::CPU> z;
+  matrix::Matrix<T, Device::CPU> ztmp;
+  matrix::Matrix<SizeType, Device::CPU> i2;
+  matrix::Matrix<ColType, Device::CPU> c;
+};
+
+template <class T>
+struct WorkSpaceHostMirror<T, Device::CPU> {
+  matrix::Matrix<T, Device::CPU>& evals;
+  matrix::Matrix<T, Device::CPU>& mat1;
+  matrix::Matrix<T, Device::CPU>& dtmp;
+  matrix::Matrix<T, Device::CPU>& z;
+  matrix::Matrix<T, Device::CPU>& ztmp;
+  matrix::Matrix<SizeType, Device::CPU>& i2;
+  matrix::Matrix<ColType, Device::CPU>& c;
 };
 
+template <class T>
+Matrix<T, Device::CPU> initMirrorMatrix(Matrix<T, Device::GPU>& mat) {
+  return Matrix<T, Device::CPU>(mat.distribution());
+}
+
+template <class T>
+Matrix<T, Device::CPU>& initMirrorMatrix(Matrix<T, Device::CPU>& mat) {
+  return mat;
+}
+
```

Comment:
I'm just trying to understand this change. Did you stop using `MatrixMirror` because it didn't give you enough control over when copies are made or something else? Just asking in case we should consider further changes to `MatrixMirror` or if it's unnecessary since the mirroring here may be removed soon enough once everything is on the GPU?

Reply:
> Did you stop using MatrixMirror because it didn't give you enough control over when copies are made or something else? 
 
 Yes. Essentially, I only needed copies for a subset of tiles [i_begin, i_end] and I didn't need a copy upon destruction.

> Just asking in case we should consider further changes to MatrixMirror or if it's unnecessary since the mirroring here may be removed soon enough once everything is on the GPU?

I don't think we need to add new functionality to MatrixMirror at the moment.